packages/realm/src/Unmanaged.ts

```diff
@@ -58,11 +56,28 @@ type OmittedRealmTypes<T> = Omit<
   | ExtractPropertyNamesOfType<T, AnyDictionary>
 >;
 
+/** Make all fields optional except those specified in K */
+type OptionalExcept<T, K extends keyof T> = Partial<T> & Pick<T, K>;
+
+/**
+ * Omits all properties of a model which are not defined by the schema,
+ * making all properties optional except those specified in RequiredProperties.
+ */
+type OmittedRealmTypesWithRequired<T, RequiredProperties extends keyof OmittedRealmTypes<T>> = OptionalExcept<
+  OmittedRealmTypes<T>,
+  RequiredProperties
+>;
+
 /** Remaps realm types to "simpler" types (arrays and objects) */
 type RemappedRealmTypes<T> = RealmListsRemappedModelPart<T> & RealmDictionaryRemappedModelPart<T>;
 
 /**
- * Joins T stripped of all keys which value extends {@link Collection} and all inherited from {@link RealmObject},
- * with only the keys which value extends {@link List}, remapped as Arrays.
+ * Joins `T` stripped of all keys which value extends {@link Collection} and all inherited from Realm.Object,
```

Comment:
Should we perhaps link or backtick this?
```suggestion
 * Joins `T` stripped of all keys which value extends {@link Collection} and all inherited from {@link Realm.Object},
```